### ISSUE TITLE
Only certain videos won't play - error message reads "video source isnt working or incorrect time"

### ISSUE THREAD

**Issue by angrykid8** [yuliskov/SmartTube#1276]
Only certain videos won't play - error message reads "video source isn't working or incorrect time"
App is up to date, nvidia shield pro 2019
videos that produce issue:
https://www.youtube.com/watch?v=eg3Jcv-a0cI
https://www.youtube.com/watch?v=nAmfCdnucWs

**Comment by rahadiancs** [comment]
Looks like related to #611 and #612

**Comment by costinmx** [comment]
Just a note: Don't have this issue on Sony Android TV and CC with Google TV. All links are playing ok with STN. Examples above included.  Playing fine.

**Comment by rahadiancs** [comment]
I do have this issue too. Not with that particular video. I noticed that most foreign video (western) doesn't load while most local region video would load just fine.

**Comment by davronthemighty** [comment]
I get this error a lot during live streams. I'm using a FireTV 4K.

**Comment by rahadiancs** [comment]
It's really weird. This issue is gone since yesterday. No app update, no nothing. It's just fixed now. Every video played without problem.

**Comment by KamauIan** [comment]
also having this same issue, only when i'm using vpn. can't play current or previous live streams.

**Comment by KamauIan** [comment]
@angrykid8 , when you get the error, check your pihole to see what was bocked, then whitelist it. the video will play.

**Comment by LeCommonMan** [comment]
**"Video source is not working or Incorrect time"**
stated to get this error past 2-3 days. Mostly on videos which has duration of over 20 minutes.
The only way to continue that video playback is - exit playback and start it again. 
Changing buffer size setting doesn't solve this.   

Device used in FireTV stick.

**Comment by SanteriHetekivi** [comment]
I have this same error on some videos my NVIDIA Shield TV Darcy (2017) (P2897) running Shield Experience 9.0.1 (Android 11) and SmartTube 14.20. I was using Pi-hole and first thing I did was to whitelist all blocked addresses. When it did not work I removed Pi-hole from my network and after that still same error. Official Youtube app and every other device with browser or Youtube app on the same network has worked fine with or without Pi-hole on same network with videos that do not work on SmartTube. This started happening couple of days ago and only for some of the videos.

**Comment by shea1337** [comment]
Found a duplicate issue: https://github.com/yuliskov/SmartTubeNext/issues/1331

I had this issue with livestreams (current OR pre-recorded)  but not videos which are not livestreams.

I have a Sony Bravia 4K VH2 TV running the latest firmware (v6.5770, Android TV OS version 10, Kernel version 4.19.75). I have tried disabling automatic date and time in the TV settings just to see if that had any effect. 

The issue started while using 14.36 and continues to happen after I updated to 14.45.

**Comment by zukapr** [comment]
I am having the same issue. What I've found out is that it happens with live streams which just finished or finished some minutes ago.

**Comment by davronthemighty** [comment]
I've had the issue on both an original firestick 4k and firecube tv (2nd gen). I've only ever had it on livestreams. I thought it was related to the buffer since most of the streamers I get it with are in other countries with possibly not the best upstream internet so I tried changing that and it doesn't impact it. In most cases it will come back after a period of seconds up to a minute. It happens both on the current stable and the current beta.

**Comment by buthig666** [comment]
Same problem with 2 Sony Bravia TVs, one A9 the other A10.  Also seems to be related to previous live streams.

**Comment by nini101** [comment]
I've been having this issue for about a month. Only happens with live streams or streams that have recently ended. Also noticed I can't rewind live streams anymore

**Comment by zukuxi** [comment]
I got this error on my Fire Stick 4k Max, but not on Fire Stick(2nd gen

**Comment by Juancollado2003** [comment]
I solved using a vpn, looks cloudflare warp is breaking something

**Comment by Salamat-UK** [comment]
Has just started happening with me, today, on my Fire Stick! The issue seems to have been reported for some time, but I don't see any definate solution? Is there one?

**Comment by vegazosten** [comment]
Is there a solution now? It's months later and I have same problems....

**Comment by c0n1c** [comment]
Hi,
I have the same issue for a while now. I used to bypass this with a VPN connection.
I have the error message on every video. I installed the latest version (16.93) on Shield 2017 with Android TV 9 on it.

I check the DNS connections and I don't blacklist anything from SmartTubeNext. But I saw 3 connections to i.ytimg.com with 30s delay. The problem could be from there, it cannot catch the flow and try it again 30s later.

Hope that could help to find a fix.

**Comment by koffeedude** [comment]
I see this thread is still open but since the latest update on both Stable and Beta I have been having this problem constantly. With each new update in the last month there is always a problem. Buffering, videos not playing, sponsorship setting not working, History not showing. When one problem gets solved a new one starts. Today its this "video source isn't working or incorrect time"  I have the latest updates and I am using Firecube TV

**Comment by c0n1c** [comment]
That's it !
I found out the cause. It was my ISP provider (a French provider called Free) which gave me a shared IPv4 address. I did a request to get a full-stack one and no error anymore.

**Comment by vegazosten** [comment]
You can fix this only after deleting the software and reinstall the new
version.

grtz<
jose

Op do 9 feb. 2023 om 09:44 schreef c0n1c ***@***.***>:

**Comment by Waldorf3** [comment]
> That's it ! I found out the cause. It was my ISP provider (a French provider called Free) which gave me a shared IPv4 address. I did a request to get a full-stack one and no error anymore.

Interesting.  I got to this page through searching for this error, and coincidentally I also have an ISP using a shared IPv4 address as gateway.  I don't see why this should cause a problem, as it is practically the same everybody with a nat-router.  My alternative is to pay another $10 per month for a dedicated IP address which I really don't need.

**Comment by CraigP3** [comment]
I have an Nvidia Shield and use Surfshark for a VPN which is on all the time.
Months ago I started experiencing the Video error randomly on a few videos but over the last 2 days (after an update) nearly half of the videos I attempt to watch have the error. So reading all of the posts here I decided to change Country from USA to Canada in my Surfshark VPN. Fixed 100% back to normal so far in a couple hours not one video has the error. I'll try it a few days and try to report back. I wonder if Traceroute might locate some corporate router that is blocking traffic? Interesting problem to solve?

**Comment by vegazosten** [comment]
Hello, can I give you a idea to make your softw better?
I missing two things.

1. It would be nice to have a windows version, can this convertion be done
with AI?
2. It would be nice  if we could use this player as a IPTV player to.
Tivimate is the best IPTVplayer ever and IPO pro Player is also great. The
rest sucks...

I know it's easter said than done but you guys should have gold if this was
possible!


Op do 20 jul 2023 om 02:16 schreef CraigP3 ***@***.***>:
